demo.zig

```diff
@@ -58,6 +57,6 @@ pub fn main() !u8 {
         std.debug.print("\t'{s}'\n", .{arg});
     }
 
-    try argsParser.printHelp(Options, std.io.getStdOut().writer());
+    try argsParser.printHelp(Options, "demo.zig", std.io.getStdOut().writer());
```

Comment:
```diff
- try argsParser.printHelp(Options, "demo.zig", std.io.getStdOut().writer());
+ try argsParser.printHelp(Options, options.executable_name orelse "demo", std.io.getStdOut().writer());
```

Reply:
Changed! 

Just out of curiosity, options.executable_name currently includes the full path to the executable. Would it be better to rename executable_name to executable_path? Or truncate to just the stem of the executable?

Reply:
> Just out of curiosity, options.executable_name currently includes the full path to the executable

This actually fully depends on the OS! Check this out:
```sh-session
[felix@denkplatte-v2 zig-args]$ ./demo
executable name: ./demo
[felix@denkplatte-v2 zig-args]$ ../zig-args/demo
executable name: ../zig-args/demo
[felix@denkplatte-v2 zig-args]$ $(realpath ../zig-args/demo )
executable name: /home/felix/projects/libraries/zig-args/demo
[felix@denkplatte-v2 zig-args]$ export "PATH=$(pwd):$PATH"
[felix@denkplatte-v2 zig-args]$ demo
executable name: demo
[felix@denkplatte-v2 zig-args]$ ln -s demo foobar
[felix@denkplatte-v2 zig-args]$ ../zig-args/foobar
executable name: ../zig-args/foobar
```

How the user processes and displays this information is up to them

Reply:
Interesting, thanks!